py/mistql/cli.py

```diff
@@ -34,20 +35,34 @@
 
 
 def main(supplied_args=None):
+    
     if supplied_args is None:
         args = parser.parse_args()
     else:
         args = parser.parse_args(supplied_args)
     raw_data: Union[str, bytes]
+    skip_json_load = False
     if args.data:
         raw_data = args.data
-    elif args.file:
-        with open(args.file, 'rb') as f:
-            raw_data = f.read()
+    elif args.file: 
+         
+        file_ext = args.file.split(".")[-1] 
+        if file_ext in ["jsonl"]: 
```

Comment:
As mentioned in the overall review, main issue is that we're relying on a file extension check.

Reply:
OK, ill modify it